### PR TITLE
Run smoke tests after prod deploy

### PR DESCRIPTION
**Pull request by cat5inthecradle**
Run integration tests against production after each deployment.

Failures are reported to #infra-javabuilder in Slack

Preview Pipeline (Delete when merging):
https://us-east-1.console.aws.amazon.com/codesuite/codepipeline/pipelines/javabuilder-smoketest-cicd/view?region=us-east-1

### CODE REVIEW

File: cicd/2-cicd/deploy-cicd.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ else
   fi
 fi
 
-ENVIRONMENT_TYPE=${ENVIRONMENT_TYPE-'standard'}
+ENVIRONMENT_TYPE=${ENVIRONMENT_TYPE-'production'}
```
Comment (molly-moen) [inline cicd/2-cicd/deploy-cicd.sh:34]: should the default be production?

**Review by molly-moen — APPROVED** [review]

File: cicd/2-cicd/deploy-cicd.sh (hunk 1 above)
Comment (molly-moen) [reply cicd/2-cicd/deploy-cicd.sh:34]: Realized this was confusing. Wondering if "adhoc" or "test" would make more sense as the default vs production

File: cicd/2-cicd/cicd.template.yml [hunk 2]
```diff
@@ -395,6 +395,28 @@ Resources:
...
+              - Name: smoke-test
+                ActionTypeId:
+                  Category: Build
+                  Owner: AWS
+                  Provider: CodeBuild
+                  Version: 1
+                InputArtifacts:
+                  - Name: sourceCode
+                Configuration:
+                  ProjectName: !Ref IntegrationTestBuildProject
```
Comment (sanchitmalhotra126) [inline cicd/2-cicd/cicd.template.yml:412]: Just want to make sure I understand, how are these different from integration tests?

File: cicd/2-cicd/deploy-cicd.sh (hunk 1 above)
Comment (cat5inthecradle) [reply cicd/2-cicd/deploy-cicd.sh:34]: We could also require it to be specified, and not provide a default. I'd like to tackle that change in a separate PR though.

File: cicd/2-cicd/cicd.template.yml (hunk 2 above)
Comment (cat5inthecradle) [reply cicd/2-cicd/cicd.template.yml:412]: The only difference is the environment variables for the domain/subdomain.

File: cicd/2-cicd/deploy-cicd.sh (hunk 1 above)
Comment (molly-moen) [reply cicd/2-cicd/deploy-cicd.sh:34]: makes sense, I'm fine with this for now

**Pull request merged**